From: Ben Hutchings <benh@debian.org>
Date: Thu, 2 Jul 2026 13:43:17 +0200
Subject: cgroup/cpuset: Fix misplaced call to reset_migrate_dl_data()

Commit f8a8358832f3 "cgroup/cpuset: Reset DL migration state on
can_attach() failure", a backport of commit 4a39eda5fdd8 upstream,
wrongly moved the call to reset_migrate_dl_data() into
cpuset_can_fork() instead of the end of cpuset_can_attach().  Move it
to the right place.

Signed-off-by: Ben Hutchings <benh@debian.org>
---
 kernel/cgroup/cpuset.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/cgroup/cpuset.c b/kernel/cgroup/cpuset.c
index 7e294d59bd1a..830546dd0768 100644
--- a/kernel/cgroup/cpuset.c
+++ b/kernel/cgroup/cpuset.c
@@ -2594,7 +2594,10 @@ static int cpuset_can_attach(struct cgroup_taskset *tset)
 	 * changes which zero cpus/mems_allowed.
 	 */
 	cs->attach_in_progress++;
+
 out_unlock:
+	if (ret)
+		reset_migrate_dl_data(cs);
 	mutex_unlock(&cpuset_mutex);
 	return ret;
 }
@@ -3398,10 +3401,7 @@ static int cpuset_can_fork(struct task_struct *task, struct css_set *cset)
 	 * changes which zero cpus/mems_allowed.
 	 */
 	cs->attach_in_progress++;
-
 out_unlock:
-	if (ret)
-		reset_migrate_dl_data(cs);
 	mutex_unlock(&cpuset_mutex);
 	return ret;
 }
